src/in_mem_graph_store.cpp

```diff
@@ -6,26 +6,238 @@
 
 namespace diskann
 {
+InMemGraphStore::InMemGraphStore(const size_t total_pts, const size_t reserve_graph_degree)
+    : AbstractGraphStore(total_pts, reserve_graph_degree)
+{
+    _graph.resize(total_pts);
+    for (size_t i = 0; i < total_pts; i++)
+    {
+        _graph[i].reserve(reserve_graph_degree);
+    }
+}
+
+std::tuple<uint32_t, uint32_t, size_t> InMemGraphStore::load(const std::string &index_path_prefix,
+                                                             const size_t num_points)
+{
+    return load_impl(index_path_prefix, num_points);
+}
+int InMemGraphStore::store(const std::string &index_path_prefix, const size_t num_points,
+                           const size_t num_frozen_points, const uint32_t start)
+{
+    return save_graph(index_path_prefix, num_points, num_frozen_points, start);
+}
+const std::vector<location_t> &InMemGraphStore::get_neighbours(const location_t i) const
+{
+    return _graph.at(i);
+}
+
+void InMemGraphStore::add_neighbour(const location_t i, location_t neighbour_id)
+{
+    _graph[i].emplace_back(neighbour_id);
+    if (_graph[i].size() > _max_observed_degree)
+    {
```

Comment:
how does this check affect perf?

Reply:
i guess add_neighbor is not called often? Also, neighbor vs neighbour make it consistent

Reply:
Yes, I will add that.